internal/stacks/stackruntime/internal/stackeval/testdata/sourcebundle/local_value/basics/local-value-basics.tfstack.hcl

```diff
@@ -0,0 +1,22 @@
+
+locals {
+  name = "jackson"
+  childName = stack.child.outputted_name
+  functional = format("Hello, %s!", "Ander")
+  mappy = {
+    name = "jackson",
+    age = 30
+  }
```

Comment:
Add the following tests:

- Add a local that references a data source (implement a dummy provider, that exposes a data source).
- Add a local that references a data source which returns an unknown value.

Reply:
@alisdair I finally got a test running for the first part of this, which basically tests whether locals work fine with data sources. Do you recall what the 'unknown value' return by the data source means here?

Reply:
I think the idea with that test was to ensure that unknowns are propagated correctly, and the local value can return an unknown. The "data source" part isn't particularly helpful, and I think we must have meant "a resource planned with unknown values"—more often that'll be a managed resource with an attribute like `"id"` or the like.

Reply:
I think the last test added is broad enough to cover both these cases then. It takes a data source, and the component outputs that value which is then stored within the local and then passed as input to another component. Should that cover the 'unknown' concept enough?